client/allocrunner/taskrunner/getter/util_linux.go

```diff
@@ -42,12 +43,35 @@ func credentials() (uint32, uint32) {
 	return userUID, userGID
 }
 
+// findHomeDir returns the home directory as provided by os.UserHomeDir. In case
+// os.UserHomeDir returns an error, we return /root if the current process is being
+// run by root, or /dev/null otherwise.
+func findHomeDir() string {
+	// When running as a systemd unit the process may not have the $HOME
+	// environment variable set, and os.UserHomeDir will return an error.
+
+	// if home is set, just use that
+	if home, err := os.UserHomeDir(); err == nil {
+		return home
+	}
+
+	// if we are the root user, use "/root"
+	if unix.Getuid() == 0 {
+		return "/root"
+	}
+
+	// nothing safe to do
+	return "/dev/null"
```

Comment:
```suggestion
	return "/nonexistent"
```

Let's use what `nobody` uses for home. I'd hate for an accidental misuse of this func (even in a test) to overwrite someone's `/dev/null` or similar. :sweat_smile: